tutorials/intellijDebugger.md

```diff
@@ -6,7 +6,140 @@ pageNav: 3
 
 # {{ title }}
 
+Before diving into tracing code with IntelliJ IDEA’s debugger, it’s essential to understand key debugger features and their purpose.
+
 This video (from LaunchCode) gives a pretty good explanation of how to use the IntelliJ IDEA debugger.
 
 @[youtube](1bCgzjatcr4)
 
+## Starting A Debugging Session
+
+To utilise the debugger, we need to make sure that the code is running in debug mode.
+
+### Run Debug Mode
+
+<box type="info" seamless>
+
+**Video reference:** [3.53 - 4.00](https://youtu.be/1bCgzjatcr4?si=fIEvUVhXHwKQeOlc&t=233): Run your program in debug mode to use debugging features.
```

Comment:
Instead of this, you can put the following nunjucks macro somewhere at the top of the page.

```js
{% macro show_video(start_m, start_s, end_m, end_s) %}
{% set start_in_seconds = (start_m | int)*60 + (start_s | int) %}
{% set end_in_seconds = (end_m | int)*60 + (end_s | int) %}

Relevant video segment ({{ start_m }}:{{ start_s }} - {{ end_m }}:{{ end_s }}):
<iframe width="560" height="315" src="https://www.youtube.com/embed/1bCgzjatcr4?si=kQCMnmebYVV1QOwu&amp;start={{ start_in_seconds }}&amp;end={{ end_in_seconds }}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
{% endmacro %}
```

After that, you can call the macro when you need to embed a video with start and end times, like so:
```
{{ show_video("1", "00", "1", "10") }}

....
{{ show_video("2", "30", "4", "00") }}
```